smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ClientConfigProperty.kt

```diff
@@ -137,13 +133,47 @@ class ClientConfigProperty private constructor(builder: Builder) {
 
         var baseClass: Symbol? = null
 
-        var required: Boolean = false
-        var constantValue: String? = null
+        var propertyType: ClientConfigPropertyType = ClientConfigPropertyType.SymbolDefault
 
         fun build(): ClientConfigProperty = ClientConfigProperty(this)
     }
 }
 
+/**
+ * Descriptor for how a configuration property is rendered when the configuration is built
+ */
+sealed class ClientConfigPropertyType {
+    /**
+     * A property type that uses the symbol type and builder symbol directly
+     */
+    object SymbolDefault : ClientConfigPropertyType()
+
+    /**
+     * Specifies that the value should be populated with a constant value that cannot be overridden in the builder.
+     * These are effectively read-only properties that will show up in the configuration type but not the builder.
+     *
+     * @param value the value to assign to the property at construction time
+     */
+    data class ConstantValue(val value: String) : ClientConfigPropertyType()
```

Comment:
**Question**: This new ADT is generally very good and I love the encapsulation it provides. It continues the pattern, however, of requiring imports necessary for the default value to be placed on the symbol itself rather than closer to the value. For example, the retry strategy default value requires six imports but the only way to bring them along is to attach them to a custom-built symbol (where otherwise a clean, global definition would be sufficient).

Is there a clean way to move these value-specific imports down into this type hierarchy?

Reply:
Personally I don't mind #1 but am fine w/ #2 and also do not like #3.

Reply:
I prefer option 2 since it more closely aligns with the usage of the default value then it does to the definition of the symbol.